### PR TITLE
WhatsApp Number Field

### PR DESCRIPTION
**Pull request by peeyush14goyal**
Issue #1298 
Added the WhatsApp field in Update Profile and Add User form

Tests:
Edit Profile
1. Invalid Number is put in Whatsapp Num field
2. Valid Number is put in Whatsapp Number field
image
image

### CODE REVIEW

File: src/Components/Users/UserProfile.tsx [hunk 1]
```diff
@@ -450,6 +466,17 @@ export default function UserProfile() {
...
                         />
                       </div>
+
+                      <div className="col-span-6 sm:col-span-3">
+                        <PhoneNumberField
+                          label="Whatsapp Number*"
+                          value={states.form.altPhoneNumber}
+                          onChange={(value: any) => [
+                            handleValueChange(value, "altPhoneNumber"),
+                          ]}
```
Comment (smartclash) [inline src/Components/Users/UserProfile.tsx]:
```suggestion
                          onChange={(value: any) => {
                            handleValueChange(value, "altPhoneNumber")
                          }}
```

**Review by smartclash — CHANGES_REQUESTED** [review]

File: src/Components/Users/UserProfile.tsx (hunk 1 above)
Comment (peeyush14goyal) [reply src/Components/Users/UserProfile.tsx]: I'll do that @smartclash

Comment (peeyush14goyal) [reply src/Components/Users/UserProfile.tsx]: Its failing when I change that. I dont know the reason. @bodhish you wrote that code. Why is it saying expression expected?

File: package.json [hunk 2]
```diff
@@ -111,7 +111,8 @@
     "re:watch": "bsb -make-world -clean-world -w",
     "build": "cross-env NODE_ENV=production ./node_modules/.bin/webpack -p --mode=production",
     "generate-build-meta": "node ./build-scripts/generate-build-version.js",
-    "test": "snyk test"
+    "test": "snyk test",
+    "cypress-test": "cross-env NODE_ENV=development npx cypress open"
```
Comment (smartclash) [inline package.json]:
```suggestion
    "test": "snyk test"
```

A PR has been made for this, https://github.com/coronasafe/care_fe/pull/1340

File: src/Components/Users/UserProfile.tsx [hunk 3]
```diff
@@ -164,8 +164,8 @@ export default function UserProfile() {
 
           if (altPhoneNumber) {
             alt_is_valid = altPhoneNumber.isValid();
-            if (alt_is_valid)
-              alt_is_valid = altPhoneNumber.getType() === "MOBILE";
+            if (alt_is_valid) console.log("Type is ", altPhoneNumber.getType());
+            alt_is_valid = altPhoneNumber.getType() === "MOBILE";
```
Comment (smartclash) [inline src/Components/Users/UserProfile.tsx]:
```suggestion
            if (alt_is_valid)
              alt_is_valid = altPhoneNumber.getType() === "MOBILE";
```

Don't leave `console.log`s

File: src/Components/Users/UserProfile.tsx [hunk 4]
```diff
@@ -190,14 +190,17 @@ export default function UserProfile() {
   const handleValueChange = (phoneNo: any, name: string) => {
     if (phoneNo && parsePhoneNumberFromString(phoneNo)?.isPossible()) {
       const form = { ...states.form };
+      console.log("Phone is ", phoneNo);
       form[name] = phoneNo;
+      console.log("FOrm is ", form);
       dispatch({ type: "set_form", form });
     }
   };
```
Comment (smartclash) [inline src/Components/Users/UserProfile.tsx]:
```suggestion
    if (phoneNo && parsePhoneNumberFromString(phoneNo)?.isPossible()) {
      const form = { ...states.form, name: phoneNo };
      dispatch({ type: "set_form", form });
    }
  };
```

Remove `console.log` here too

File: src/Components/Users/UserProfile.tsx [hunk 5]
```diff
@@ -190,14 +190,17 @@ export default function UserProfile() {
...
 
   const handleSubmit = async (e: any) => {
     e.preventDefault();
     const validForm = validateForm();
+    console.log("Valid form is ", validForm);
```
Comment (smartclash) [inline src/Components/Users/UserProfile.tsx]:
```suggestion
```
And this too

**Review by smartclash — CHANGES_REQUESTED** [review]

**Comment by aparnacoronasafe** [comment]
In the "create new user " form, the minute i enter DOB, the page becomes blank

File: src/Components/Users/UserProfile.tsx (hunk 1 above)
Comment (bodhish) [reply src/Components/Users/UserProfile.tsx]: @peeyush14goyal Not sure. Will have to dig it. It would mostly be a refactor that is showing my name

**Review by bodhish — CHANGES_REQUESTED** [review]
@peeyush14goyal can you refactor

File: src/Components/Users/UserProfile.tsx (hunk 1 above)
Comment (peeyush14goyal) [reply src/Components/Users/UserProfile.tsx]: Changed to { } @bodhish @smartclash

File: package.json (hunk 2 above)
Comment (peeyush14goyal) [reply package.json]: Merged from develop branch @smartclash

File: src/Components/Users/UserProfile.tsx (hunk 3 above)
Comment (peeyush14goyal) [reply src/Components/Users/UserProfile.tsx]: Removed `console.log`

File: src/Components/Users/UserProfile.tsx (hunk 4 above)
Comment (peeyush14goyal) [reply src/Components/Users/UserProfile.tsx]: All `console.log` removed

File: src/Components/Users/UserProfile.tsx (hunk 5 above)
Comment (peeyush14goyal) [reply src/Components/Users/UserProfile.tsx]: Removed @smartclash

**Comment by peeyush14goyal** [comment]
@aparnacoronasafe I have fixed the date issue.

**Comment by aparnacoronasafe** [comment]
tested in preview

**Pull request merged**